spark2/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -123,6 +123,13 @@
     this.partitionedFanoutEnabled = options.getBoolean(SparkWriteOptions.FANOUT_ENABLED, tablePartitionedFanoutEnabled);
   }
 
+  protected JavaSparkContext lazySparkContext() {
```

Comment:
Out of curiosity, why do we need a lazy one?

And also what the java one provide?

Reply:
W.r.t. lazy, I just copied the old behavior that probably saves an instance creation in case if don't get to broadcasting and fail early. It is probably be alright to always construct it, I guess.

Reply:
I took one more look and I think we should be fine removing the laziness.

@RussellSpitzer @rdblue, do you agree?

Reply:
I went ahead and removed the laziness. I think it should be safe.